### DISCUSSION TITLE
Error 404 but record does exist

### DISCUSSION THREAD

**Discussion by hugoschreib** [pocketbase/pocketbase#2307] [Q&A] [answered]
Hi, I want to update data from a auth collection using this code
`const record = await pb.collection('users').update($currentUser.id,{someField: "someValue"});`
In my case `$currentUser.id` is "fenad9780q7qqq8" when i run the code.
However, it return error 404 as if the user is not existing.
The strange part is if i run `await pb.collection('users').getOne($currentUser.id);`, it returns the correct user

![image](https://user-images.githubusercontent.com/44979019/232370088-babee477-eaff-421d-9db0-adc55443c8b4.png)

![image](https://user-images.githubusercontent.com/44979019/232370223-93e01a43-8684-4b3c-b6f8-f4fdc44c86f3.png)

**Comment by ganigeorgiev** [comment, marked as answer]
Make sure that your client request satisfies the "Update API rule" of the `users` collection (_to rule out that the rule is not the issue, you can temporary leave it empty/blank to allow everyone access_).

**Reply by jimafisk** [reply]
Hi @ganigeorgiev, I ran into a similar issue today that was exactly this (being denied by the API rule). It was confusing to debug because like @hugoschreib I could see the record in admin UI.

Would it make sense to return a `403 access denied` instead of a `404 not found` if an API rule is not being met?

**Reply by ganigeorgiev** [reply]
This seems to be a common error (https://github.com/pocketbase/pocketbase/discussions/1570, https://github.com/pocketbase/pocketbase/discussions/256#discussioncomment-3302473m, https://github.com/pocketbase/pocketbase/discussions/1046, etc.) but unfortunately it is a little more complicated as the API rules act also as admin-level filters and are executed together with the main query. At the moment there is no easy and performant way (_without extra parsing and some sort of static analysis_) to detect which exact part of the rule caused the query to fail.

**Reply by jimafisk** [reply]
Thanks for the quick reply, makes sense!
